compiler/noirc_frontend/src/hir/type_check/errors.rs

```diff
@@ -227,6 +232,15 @@ impl<'a> From<&'a TypeCheckError> for Diagnostic {
                     *expr_span,
                 )
             }
+            // TODO(https://github.com/noir-lang/noir/issues/6238): implement
+            // handling for larger types
+            TypeCheckError::EvaluatedGlobalPartialSizeChecks { expected_kind, expr_kind, expr_span } => {
+                Diagnostic::simple_warning(
+                    format!("Expected type {expected_kind:?} when evaluating globals: found {expr_kind:?}, size checks are being implemented for this case"),
```

Comment:
```suggestion
                    format!("Expected type `{expected_kind}` when evaluating global, found `{expr_kind}`"),
```

Reply:
I don't think we need to - we can add future errors or warnings for this case still

Reply:
The colon still reads oddly to me. Also - do we not have a Display impl for kinds?

Reply:
Updated to `"Expected type {expected_kind} when evaluating globals, but found {expr_kind} (this warning may become an error in the future)"`